posts/2021-04-01-open-liberty-devfile-stack-minikube.adoc

```diff
@@ -0,0 +1,127 @@
+
+Check out a quick demo:
+++++
+<iframe width="560" height="315" align="center" src="https://www.youtube.com/watch?v=KFjVGPyL49Q" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+++++
+
+== Minikube
+
+link:https://minikube.sigs.k8s.io/docs/[Minikube] is Kubernetes running right on your laptop. It's easy to install, fast to start up, and even comes with a web-based dashboard for viewing and managing your deployments. 
+
+== Open Liberty devfile stack
+
+The link:https://github.com/OpenLiberty/application-stack#open-liberty-application-stack[Open Liberty devfile stack] provides much of the tooling and infrastructure you need to start developing your application. Open Liberty, Maven, Open J9, etc. all come packaged in a development image where your app is built and run. A fully configured link:https://docs.devfile.io/devfile/2.0.0/user-guide/index.html[devfile] describes the dev environment and defines all of the build/run commands behind the scenes. You can just sit back and focus on your code. 
+
+== OpenShift Do (odo)
+
+Odo is a simple CLI tool that allows you to create stack projects and interact directly with your Kubernetes cluster (e.g. Minikube). Once you link:https://odo.dev/docs/installing-odo/[install odo], you can start creating Open Liberty stack projects and developing right in Kubernetes.
+
+== Try it out
+
+After link:https://minikube.sigs.k8s.io/docs/start/[installing Minikube], start your local cluster:
+
+[source,sh]
+----
+$ minikube start
+----
+
+MacOS users may need to start Minikube with a driver other than "docker" due to a known issue with the docker driver and the ingress addon. For example: `--driver=virtualbox`
+
+Next, enable the Minikube ingress addon:
+
+[source,sh]
+----
+$ minikube addons enable ingress
+----
+
+This will allow you to create a URL for your app using an ingress domain.
+
+Start the Minikube dashboard (Optional):
+
+[source,sh]
+----
+$ minikube dashboard
+----
+
+Minikube will open a k8s dashboard in your browser. While not necessary, it allows you to view and interact with your cluster and your development pod. 
+
+
+With Minikube set up, you are now ready to create an Open Liberty stack project. You can either initialize an existing application or use the link:https://github.com/OpenLiberty/application-stack-starters[stack starter] to create a new app. For this blog, we are going to use the link:https://github.com/OpenLiberty/application-stack-intro/tree/blog-demo[application-stack-intro] app. This application is a basic JAX-RS sample that uses Maven, Open Liberty, and MicroProfile.
+
+[source,sh]
+----
+$ git clone -b blog-demo git@github.com:OpenLiberty/application-stack-intro.git && \
+  cd application-stack-intro
+----
+
+Initialize the app as a "java-openliberty" stack project:
+
+[source,sh]
```

Comment:
Understand leaving the project/NS out of the video but maybe we could add it back here in the written directions?

Reply:
Good point. Added that back in.